ev-11

KristiyanNley opened the pull request:
Implement:
Feedback
FeedbackRequestDto
addFeedback
Add unit tests:
GivenExistingEvent_WhenAddingFeedback_FeedbackIsAdded GivenNonExistingEvent_WhenAddingFeedback_ThrowsException

KristiyanNley committed:
ev-11
Implement:
Feedback
FeedbackRequestDto
addFeedback
Add unit tests:
GivenExistingEvent_WhenAddingFeedback_FeedbackIsAdded
GivenNonExistingEvent_WhenAddingFeedback_ThrowsException

Yotzovv commented on eventorganizer.api/src/main/java/com/event/organizer/api/model/dto/FeedbackRequestDto.java:
```diff
@@ -0,0 +1,33 @@
+    private String comment;
+    private Long eventId;
+
+    public Integer setBoundedRating(Integer rating) {
```
- This should be attached to the Setter of "rating" so you need to rename "setBoundedRating" to "setRating" 
- You don't need to wrap the `boundedRating = rating` in an `else { }` as it will be hit anyway.
- You also don't need the `boundedRating` variable
- Make it "void" instead and directly assign `this.rating=rating`

so your method should look like this

```

public void setRating(Integer rating) {
        this.rating = rating;
        
        if (rating >= max) {
            this.rating = max;
        }
        else if (rating <= min) {
            this.rating = min;
        }
    }
```

Yotzovv replied:
Just checked how to do this in Java, you only need to rename "setBoundedRating" to "setRating"

KristiyanSelveliev commented on eventorganizer.api/src/main/java/com/event/organizer/api/service/EventService.java:
```diff
@@ -138,4 +139,27 @@ public void addImage(String image, Long eventId, String username) throws EventOr
+        }
+
+        Event event = eventRepository.findById(eventId).get();
+        var allFeedbacks = event.getFeedbacks();
```
Use List<Feedback> instead of var. It is bad practice to use var in java

KristiyanSelveliev commented on eventorganizer.api/src/main/java/com/event/organizer/api/model/dto/FeedbackRequestDto.java:
```diff
@@ -0,0 +1,33 @@
+@Getter
+@Setter
+public class FeedbackRequestDto {
+    private final Integer min = 1;
```
MIN and MAX should be constants 
private static final Integer MIN = 1;
private static final Integer MAX = 5;

KristiyanNley committed:
ev-9
Remove unused variable boundedRating
Change setBoundedRating to setRating
Change var to List<*>
Make constants static
